src/rt/backtrace/dwarf.d

```diff
@@ -206,7 +260,42 @@ void resolveAddresses(const(ubyte)[] debugLineSectionData, Location[] locations,
     }
 }
 
+/**
+ * A callback type for `runStateMachine`
+ *
+ * The callback is called when certain specific opcode are encountered
+ * (a.k.a when a complete `LocationInfo` is encountered).
+ * See `runStateMachine` implementation and the DWARF specs for more detail.
+ *
+ * Params:
+ *   address = The address that the `LocationInfo` describes
+ *   info = The `LocationInfo` itself, describing `address`
+ *   isEndSequence = Whether the end of a sequence has been reached
+ */
 alias RunStateMachineCallback = bool delegate(size_t, LocationInfo, bool) @nogc nothrow;
+
+/**
+ * Run the state machine to generate line number matrix
+ *
+ * Line number informations generated by the compiler are stored in the
+ * `.debug_line` section. Conceptually, they can be seen as a large matrix,
+ * with row such as "file", "line", "column", "is_statement", etc...
+ * However such a matrix would be too big to store in an object file,
+ * so DWARF instead generate this matrix using bytecode fed to a state machine.
+ *
+ * Note:
+ * Each compilation unit (module in D) ccan have its own line number program.
```

Comment:
Little typo ('ccan'); and a compilation unit isn't necessarily a D module.

Reply:
Thanks, fixed